### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by Nessrine88**
In this project, I have fulfilled the following requirements:

- Added RSpec as a gem to the project.
- Created a Request spec file for all controllers.
- Ensured that for each controller action, I checked the following:
- If the response status was correct.
- If the correct template was rendered.
- If the response body included the correct placeholder text

### CODE REVIEW

File: spec/requests/users_spec.rb
Comment (aamirkhan2478) [inline spec/requests/users_spec.rb:1]: - [ ] Please make sure all your specs should be passed but for now, some of your specs are not passed so kindly fix your specs check below :point_down: the screenshot of the failed specs :slightly_smiling_face: 

Screenshot from 2023-11-08 21-38-35

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @Nessrine88, 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+describe 'GET /index' do
+  it 'returns a successful response' do
+    user = User.create!(name: 'User1')
+    get "/users/#{user.id}/posts/index"
+    expect(response).to be_successful
+  end
```
Comment (abdulhamiid) [inline spec/requests/posts_spec.rb:8]: image

The issue at hand is that the error message "Posts counter is not a number" indicates that in order to create a new `user`, our validation process necessitates the presence of an initialized integer value for the `posts_counter` field. 

We can address this by ensuring that `posts_counter` is appropriately initialized as an integer. 🙂💪

```sql
user = User.create!(name: 'User1', posts_counter: 0)
```

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Nessrine88 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, I really appreciate the effort you put in! but there are still some which haven't been addressed yet.

almost-there

## Highlights 🌟

- ✔️ Write tests for controllers
- ✔️ Use Git Flow

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @Nessrine88 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All tests are passing   ✅ 
  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**